whois-nrtm4/src/main/java/net/ripe/db/nrtm4/generator/SnapshotFileGenerator.java

```diff
@@ -82,47 +85,59 @@ public void createSnapshot()  {
         LOGGER.info("Snapshot generation complete {}", stopwatch);
+
         cleanUpOldFiles();
     }
 
-    private void saveToDatabase(final List<NrtmVersionInfo> sourceToNewVersion,  final Map<CIString, byte[]> payloadBySource) {
-        payloadBySource.forEach( (source, payload ) -> {
-            final Optional<NrtmVersionInfo> versionInfo = sourceToNewVersion.stream().filter( (version) -> version.source().getName().equals(source)).findAny();
-            if(versionInfo.isPresent()) {
-                try {
-                    final String fileName = NrtmFileUtil.newGzFileName(versionInfo.get());
-                    LOGGER.info("Source {} snapshot file {}", source, fileName);
-                    LOGGER.info("Calculated hash for {}", source);
-                    nrtmFileRepository.saveSnapshotVersion(versionInfo.get(), fileName, calculateSha256(payload), payload);
-                    LOGGER.info("Wrote {} to DB {}", source);
-                } catch (final Throwable t) {
-                    LOGGER.error("Unexpected throwable caught when inserting snapshot file", t);
+    private Map<CIString, byte[]> writeToGzipStream(final SnapshotState snapshotState, final List<NrtmVersionInfo> sourceToNewVersion) {
+        final Map<CIString, GzipOutStreamWriter> sourceResources = initializeResources(sourceToNewVersion);
+
+        final AtomicInteger noOfBatchesProcessed = new AtomicInteger(0);
+        final List<List<WhoisObjectData>> batches = Lists.partition(snapshotState.whoisObjectData(), BATCH_SIZE);
+
+        final Timer timer = new Timer(true);
+        printProgress(noOfBatchesProcessed, snapshotState.whoisObjectData().size(), timer);
+
+        try {
+            batches.parallelStream().map(objectBatch -> {
+                final List<RpslObject> rpslObjects = Lists.newArrayList();
+
+                whoisObjectRepository.findRpslMapForObjects(objectBatch).values().forEach( object -> {
+                    try {
+                        final RpslObject rpslObject = RpslObject.parse(object);
+                        if (dummifierNrtmV4.isAllowed(rpslObject)) {
+                            rpslObjects.add(dummifierNrtmV4.dummify(rpslObject));
+                        }
+                    } catch (final Exception e) {
+                        LOGGER.warn("Parsing RPSL threw exception", e);
+                    }
+                });
+
+                noOfBatchesProcessed.incrementAndGet();
+                return rpslObjects;
+            }).flatMap(Collection::stream).forEach(rpslObject -> {
+                if(sourceResources.containsKey(rpslObject.getValueForAttribute(AttributeType.SOURCE))) {
+                    sourceResources.get(rpslObject.getValueForAttribute(AttributeType.SOURCE)).write(new SnapshotFileRecord(rpslObject));
                 }
-            }
-        });
+            });
+
+        } catch (final Exception e) {
+            LOGGER.warn("Error while writing snapshotfile", e);
```

Comment:
why is this a warning and not an error?

Reply:
Runtime exception is thrown after that which will anyways be reported and logged. Will change to error anyways